### PR TITLE
dev-python/hypothesis: bump to 3.6.0, eapi bump, fixed dependencies

### PR DESCRIPTION
**Pull request by MathyV**
A new version was released that removes the dev-python/uncompyle6 dep
introduced in 3.5.3. This also means that the KEYWORDS and pypy3 support
that were removed in eea41c4ca5bcd27af35f3276459567b91e4a34e7 can be
reintroduced. Also 3.4.2 no longer needs to be kept in the tree.

This package has a dependency on python being built with sqlite support.
It isn't obvious and only shows when other packages (like cryptography)
use certain modules of the library.

Gentoo-Bug: https://bugs.gentoo.org/show_bug.cgi?id=593824
Gentoo-Bug: https://bugs.gentoo.org/show_bug.cgi?id=579514
Gentoo-Bug: https://bugs.gentoo.org/show_bug.cgi?id=580540

Package-Manager: portage-2.3.0

### CODE REVIEW

File: dev-python/hypothesis/hypothesis-3.6.0.ebuild [hunk 1]
```diff
@@ -15,12 +15,11 @@ SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
 
 LICENSE="MPL-2.0"
 SLOT="0"
-KEYWORDS="~amd64 ~x86"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86"
 IUSE=""
 
 DEPEND="dev-python/setuptools[${PYTHON_USEDEP}]
 	$(python_gen_cond_dep 'dev-python/enum34[${PYTHON_USEDEP}]' python2_7)
```
Comment (SoapGentoo) [inline dev-python/hypothesis/hypothesis-3.6.0.ebuild]: please change the `python2_7` to `'python2*'`

File: dev-python/hypothesis/hypothesis-3.6.0.ebuild [hunk 2]
```diff
@@ -15,12 +15,11 @@ SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
...
-	dev-python/uncompyle6[${PYTHON_USEDEP}]
 "
 RDEPEND="${DEPEND}"
```
Comment (SoapGentoo) [inline dev-python/hypothesis/hypothesis-3.6.0.ebuild]: `optfeature` requires `eutils`

File: dev-python/hypothesis/hypothesis-3.6.0.ebuild [hunk 3]
```diff
@@ -15,12 +15,11 @@ SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
...
```
Comment (SoapGentoo) [inline dev-python/hypothesis/hypothesis-3.6.0.ebuild]: swap definitions of `RDEPEND` and `DEPEND`

File: dev-python/hypothesis/hypothesis-3.6.0.ebuild (hunk 1 above)
Comment (MathyV) [inline dev-python/hypothesis/hypothesis-3.6.0.ebuild]: Done

File: dev-python/hypothesis/hypothesis-3.6.0.ebuild (hunk 3 above)
Comment (MathyV) [inline dev-python/hypothesis/hypothesis-3.6.0.ebuild]: Done

File: dev-python/hypothesis/hypothesis-3.6.0.ebuild (hunk 2 above)
Comment (MathyV) [inline dev-python/hypothesis/hypothesis-3.6.0.ebuild]: Done

**Pull request closed without merging**